pkg/podman/errors.go

```diff
@@ -0,0 +1,17 @@
+package podman
+
+import (
+	"fmt"
+)
+
+type PodmanNotFoundError struct {
+	err error
+}
+
+func NewPodmanNotFoundError(err error) PodmanNotFoundError {
+	return PodmanNotFoundError{err: err}
+}
+
+func (o PodmanNotFoundError) Error() string {
+	return fmt.Errorf("unable to access podman. Do you have podman client installed? Cause: %w", o.err).Error()
```

Comment:
Can `o.err` be nil? In this case, the string returned here might need to be changed.

For example, when running `odo describe component` (which should describe the component on both Podman and the cluster), I got the following message:
```
$ ODO_EXPERIMENTAL_MODE=t PODMAN_CMD=404 odo describe component
============================================================================
⚠ Experimental mode enabled. Use at your own risk.
More details on https://odo.dev/docs/user-guides/advanced/experimental-mode
============================================================================

 ⚠  unable to access podman. Do you have podman client installed? Cause: %!w(<nil>)
 ⚠  failed to get ingresses/routes: ingresses.networking.k8s.io is forbidden: User "system:anonymous" cannot list resource "ingresses" in API group "networking.k8s.io" in the namespace "rh-ee-asoro-dev"
Name: bug-odo-logs-no-cluster
Display Name: Go Runtime
Project Type: Go
Language: Go
Version: 1.0.2
Description: Go is an open source programming language that makes it easy to build simple, reliable, and efficient software.
Tags: Go

Running in: None

Supported odo features:
 •  Dev: true
 •  Deploy: false
 •  Debug: false

Container components:
 •  runtime
    Source Mapping: /projects
```

Reply:
> 

I think you raised good points, but I think they might be out of scope for this PR. The errors "executable %q not found" is raised when you initiliaze a new Podman Client. This PR is more about logging error when there is one. I'd be happy to work on it in a separate PR.